References
==========

Chakravarty, Debyani, et al. “OncoKB: A Precision Oncology Knowledge Base.” JCO Precision Oncology, no. 1, 2017, pp. 1–16., https://doi.org/10.1200/po.17.00011. 

COSMIC Cancer Gene Census https://cancer.sanger.ac.uk/census

Ellrott, Kyle, Matthew H. Bailey, Gordon Saksena, Kyle R. Covington, Cyriac Kandoth, Chip Stewart, Julian Hess, et al. 2018. “Scalable Open Science Approach for Mutation Calling of Tumor Exomes Using Multiple Genomic Pipelines.” Cell Systems 6 (3): 271–81.e7.

Gao, Qingsong, Wen-Wei Liang, Steven M. Foltz, Gnanavel Mutharasu, Reyka G. Jayasinghe, Song Cao, Wen-Wei Liao, et al. 2018. “Driver Fusions and Their Implications in the Development and Treatment of Human Cancers.” Cell Reports 23 (1): 227–38.e3.

Knijnenburg, Theo A., Linghua Wang, Michael T. Zimmermann, Nyasha Chambwe, Galen F. Gao, Andrew D. Cherniack, Huihui Fan, et al. 2018. “Genomic and Molecular Landscape of DNA Damage Repair Deficiency across The Cancer Genome Atlas.” Cell Reports 23 (1): 239–54.e6.

Mayakonda, Anand, De-Chen Lin, Yassen Assenov, Christoph Plass, and H. Phillip Koeffler. 2018. “Maftools: Efficient and Comprehensive Analysis of Somatic Variants in Cancer.” Genome Research 28 (11): 1747–56.

“MSK's Precision Oncology Knowledge Base.” OncoKB, https://www.oncokb.org/. 

UCSC Xena TCGA Pan-Cancer Atlas Data Hub https://xenabrowser.net/datapages/?hub=https://pancanatlas.xenahubs.net:443

Griffith, M., Spies, N., Krysiak, K. et al. CIViC is a community knowledgebase for expert crowdsourcing the clinical interpretation of variants in cancer. Nat Genet 49, 170–174 (2017). https://doi.org/10.1038/ng.3774
