{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using the CIViC API"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import api_tools as apit"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "### Load Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>description</th>\n",
       "      <th>flagged</th>\n",
       "      <th>variants</th>\n",
       "      <th>aliases</th>\n",
       "      <th>updated_at</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>ALK amplifications, fusions and mutations have...</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'CAD-ALK', 'id': 2769, 'evidence_ite...</td>\n",
       "      <td>[ALK1, NBLST3, ALK, CD246]</td>\n",
       "      <td>2017-03-06T00:00:15.358Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>AKT1, also referred to as protein kinase B, is...</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'Overexpression', 'id': 2360, 'evide...</td>\n",
       "      <td>[AKT1, RAC-ALPHA, RAC, PRKBA, PKB-ALPHA, AKT, ...</td>\n",
       "      <td>2017-02-09T21:58:06.184Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ARAF</td>\n",
       "      <td>369</td>\n",
       "      <td>ARAF has recently become increasingly consider...</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'S490T', 'id': 825, 'evidence_items'...</td>\n",
       "      <td>[ARAF1, ARAF, RAFA1, PKS2, A-RAF]</td>\n",
       "      <td>2015-06-21T16:49:19.894Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>ABL1 is most relevant to cancer in its role in...</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'BCR-ABL Q252H', 'id': 1024, 'eviden...</td>\n",
       "      <td>[JTK7, bcr/abl, ABL1, v-abl, c-ABL1, CHDSKM, p...</td>\n",
       "      <td>2015-06-21T16:49:20.159Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>673</td>\n",
       "      <td>BRAF mutations are found to be recurrent in ma...</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'ZKSCAN1-BRAF', 'id': 657, 'evidence...</td>\n",
       "      <td>[BRAF-1, BRAF, RAFB1, B-raf, BRAF1, NS7, B-RAF1]</td>\n",
       "      <td>2019-09-19T19:32:44.942Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>5177</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>6390</td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'p.193_196dupSTSC (c.577_588dupAGCAC...</td>\n",
       "      <td>[SDH1, PGL4, SDH2, SDH, SDHIP, CWS2, MC2DN4, S...</td>\n",
       "      <td>2017-02-09T22:00:31.813Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>5210</td>\n",
       "      <td>SRSF2</td>\n",
       "      <td>6427</td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'MUTATION', 'id': 595, 'evidence_ite...</td>\n",
       "      <td>[SRSF2, SRp30b, SFRS2A, SFRS2, SC35, SC-35, PR...</td>\n",
       "      <td>2017-02-09T22:00:32.685Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>5225</td>\n",
       "      <td>SGK1</td>\n",
       "      <td>6446</td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'Overexpression', 'id': 693, 'eviden...</td>\n",
       "      <td>[SGK1, SGK]</td>\n",
       "      <td>2015-09-03T20:30:22.694Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>5356</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>6598</td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'UNDEREXPRESSION', 'id': 695, 'evide...</td>\n",
       "      <td>[CSS3, INI-1, SMARCB1, Sfh1p, PPP1R144, SWNTS1...</td>\n",
       "      <td>2015-09-03T20:30:55.169Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>5365</td>\n",
       "      <td>SMO</td>\n",
       "      <td>6608</td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>[{'name': 'D473H', 'id': 299, 'evidence_items'...</td>\n",
       "      <td>[Gx, PHLS, SMOH, SMO, FZD11, CRJS]</td>\n",
       "      <td>2015-09-03T20:31:02.628Z</td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>238 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       id     name  entrez_id  \\\n",
       "0       1      ALK        238   \n",
       "1       2     AKT1        207   \n",
       "2       3     ARAF        369   \n",
       "3       4     ABL1         25   \n",
       "4       5     BRAF        673   \n",
       "..    ...      ...        ...   \n",
       "233  5177     SDHB       6390   \n",
       "234  5210    SRSF2       6427   \n",
       "235  5225     SGK1       6446   \n",
       "236  5356  SMARCB1       6598   \n",
       "237  5365      SMO       6608   \n",
       "\n",
       "                                           description  flagged  \\\n",
       "0    ALK amplifications, fusions and mutations have...    False   \n",
       "1    AKT1, also referred to as protein kinase B, is...    False   \n",
       "2    ARAF has recently become increasingly consider...    False   \n",
       "3    ABL1 is most relevant to cancer in its role in...    False   \n",
       "4    BRAF mutations are found to be recurrent in ma...    False   \n",
       "..                                                 ...      ...   \n",
       "233                                                       False   \n",
       "234                                                       False   \n",
       "235                                                       False   \n",
       "236                                                       False   \n",
       "237                                                       False   \n",
       "\n",
       "                                              variants  \\\n",
       "0    [{'name': 'CAD-ALK', 'id': 2769, 'evidence_ite...   \n",
       "1    [{'name': 'Overexpression', 'id': 2360, 'evide...   \n",
       "2    [{'name': 'S490T', 'id': 825, 'evidence_items'...   \n",
       "3    [{'name': 'BCR-ABL Q252H', 'id': 1024, 'eviden...   \n",
       "4    [{'name': 'ZKSCAN1-BRAF', 'id': 657, 'evidence...   \n",
       "..                                                 ...   \n",
       "233  [{'name': 'p.193_196dupSTSC (c.577_588dupAGCAC...   \n",
       "234  [{'name': 'MUTATION', 'id': 595, 'evidence_ite...   \n",
       "235  [{'name': 'Overexpression', 'id': 693, 'eviden...   \n",
       "236  [{'name': 'UNDEREXPRESSION', 'id': 695, 'evide...   \n",
       "237  [{'name': 'D473H', 'id': 299, 'evidence_items'...   \n",
       "\n",
       "                                               aliases  \\\n",
       "0                           [ALK1, NBLST3, ALK, CD246]   \n",
       "1    [AKT1, RAC-ALPHA, RAC, PRKBA, PKB-ALPHA, AKT, ...   \n",
       "2                    [ARAF1, ARAF, RAFA1, PKS2, A-RAF]   \n",
       "3    [JTK7, bcr/abl, ABL1, v-abl, c-ABL1, CHDSKM, p...   \n",
       "4     [BRAF-1, BRAF, RAFB1, B-raf, BRAF1, NS7, B-RAF1]   \n",
       "..                                                 ...   \n",
       "233  [SDH1, PGL4, SDH2, SDH, SDHIP, CWS2, MC2DN4, S...   \n",
       "234  [SRSF2, SRp30b, SFRS2A, SFRS2, SC35, SC-35, PR...   \n",
       "235                                        [SGK1, SGK]   \n",
       "236  [CSS3, INI-1, SMARCB1, Sfh1p, PPP1R144, SWNTS1...   \n",
       "237                 [Gx, PHLS, SMOH, SMO, FZD11, CRJS]   \n",
       "\n",
       "                   updated_at  type  \n",
       "0    2017-03-06T00:00:15.358Z  gene  \n",
       "1    2017-02-09T21:58:06.184Z  gene  \n",
       "2    2015-06-21T16:49:19.894Z  gene  \n",
       "3    2015-06-21T16:49:20.159Z  gene  \n",
       "4    2019-09-19T19:32:44.942Z  gene  \n",
       "..                        ...   ...  \n",
       "233  2017-02-09T22:00:31.813Z  gene  \n",
       "234  2017-02-09T22:00:32.685Z  gene  \n",
       "235  2015-09-03T20:30:22.694Z  gene  \n",
       "236  2015-09-03T20:30:55.169Z  gene  \n",
       "237  2015-09-03T20:31:02.628Z  gene  \n",
       "\n",
       "[238 rows x 9 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "civic_gene = apit.Endpoint(url=\"https://civicdb.org/api/genes?count=238\")\n",
    "df_civic_gene = civic_gene.data_as_pandas(\"records\")\n",
    "df_civic_gene"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>entrez_name</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>type</th>\n",
       "      <th>variant_types</th>\n",
       "      <th>civic_actionability_score</th>\n",
       "      <th>flagged</th>\n",
       "      <th>updated_at</th>\n",
       "      <th>coordinates</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR-ABL</td>\n",
       "      <td>The BCR-ABL fusion protein, commonly referred ...</td>\n",
       "      <td>4</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 120, 'name': 'transcript_fusion', 'dis...</td>\n",
       "      <td>255.5</td>\n",
       "      <td>False</td>\n",
       "      <td>2020-02-08T03:53:21.874Z</td>\n",
       "      <td>{'chromosome': '22', 'start': 23522397, 'stop'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR-ABL T315I</td>\n",
       "      <td>While the efficacy of imatinib has revolutioni...</td>\n",
       "      <td>4</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>146.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2020-08-29T00:06:08.064Z</td>\n",
       "      <td>{'chromosome': '9', 'start': 133748283, 'stop'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR-ABL E255K</td>\n",
       "      <td>While the efficacy of imatinib has revolutioni...</td>\n",
       "      <td>4</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>83.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2021-06-08T23:03:08.149Z</td>\n",
       "      <td>{'chromosome': '9', 'start': 133738363, 'stop'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>E17K</td>\n",
       "      <td>AKT1 E17K is a recurrent mutation that has bee...</td>\n",
       "      <td>2</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>33.5</td>\n",
       "      <td>False</td>\n",
       "      <td>2021-11-12T18:53:34.889Z</td>\n",
       "      <td>{'chromosome': '14', 'start': 105246551, 'stop...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>EML4-ALK</td>\n",
       "      <td>The EML4-ALK fusion variant 1 consisting of AL...</td>\n",
       "      <td>1</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 120, 'name': 'transcript_fusion', 'dis...</td>\n",
       "      <td>38.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2020-08-29T19:17:30.276Z</td>\n",
       "      <td>{'chromosome': '2', 'start': 42396490, 'stop':...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3051</th>\n",
       "      <td>3696</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>2261</td>\n",
       "      <td>R669G</td>\n",
       "      <td></td>\n",
       "      <td>23</td>\n",
       "      <td>variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2022-01-07T06:56:37.814Z</td>\n",
       "      <td>{'chromosome': None, 'start': None, 'stop': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3052</th>\n",
       "      <td>3697</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>2261</td>\n",
       "      <td>R669Q</td>\n",
       "      <td></td>\n",
       "      <td>23</td>\n",
       "      <td>variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2022-01-07T06:57:07.685Z</td>\n",
       "      <td>{'chromosome': None, 'start': None, 'stop': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3053</th>\n",
       "      <td>3698</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>2261</td>\n",
       "      <td>D617G</td>\n",
       "      <td></td>\n",
       "      <td>23</td>\n",
       "      <td>variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2022-01-07T06:58:48.728Z</td>\n",
       "      <td>{'chromosome': None, 'start': None, 'stop': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3054</th>\n",
       "      <td>3699</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>2261</td>\n",
       "      <td>G637W</td>\n",
       "      <td></td>\n",
       "      <td>23</td>\n",
       "      <td>variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2022-01-07T06:59:38.140Z</td>\n",
       "      <td>{'chromosome': None, 'start': None, 'stop': No...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3055</th>\n",
       "      <td>3700</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>4914</td>\n",
       "      <td>ETV6-NTRK1</td>\n",
       "      <td></td>\n",
       "      <td>3983</td>\n",
       "      <td>variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2022-01-07T20:57:08.903Z</td>\n",
       "      <td>{'chromosome': None, 'start': None, 'stop': No...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3056 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        id entrez_name  entrez_id           name  \\\n",
       "0        1        ABL1         25        BCR-ABL   \n",
       "1        2        ABL1         25  BCR-ABL T315I   \n",
       "2        3        ABL1         25  BCR-ABL E255K   \n",
       "3        4        AKT1        207           E17K   \n",
       "4        5         ALK        238       EML4-ALK   \n",
       "...    ...         ...        ...            ...   \n",
       "3051  3696       FGFR3       2261          R669G   \n",
       "3052  3697       FGFR3       2261          R669Q   \n",
       "3053  3698       FGFR3       2261          D617G   \n",
       "3054  3699       FGFR3       2261          G637W   \n",
       "3055  3700       NTRK1       4914     ETV6-NTRK1   \n",
       "\n",
       "                                            description  gene_id     type  \\\n",
       "0     The BCR-ABL fusion protein, commonly referred ...        4  variant   \n",
       "1     While the efficacy of imatinib has revolutioni...        4  variant   \n",
       "2     While the efficacy of imatinib has revolutioni...        4  variant   \n",
       "3     AKT1 E17K is a recurrent mutation that has bee...        2  variant   \n",
       "4     The EML4-ALK fusion variant 1 consisting of AL...        1  variant   \n",
       "...                                                 ...      ...      ...   \n",
       "3051                                                          23  variant   \n",
       "3052                                                          23  variant   \n",
       "3053                                                          23  variant   \n",
       "3054                                                          23  variant   \n",
       "3055                                                        3983  variant   \n",
       "\n",
       "                                          variant_types  \\\n",
       "0     [{'id': 120, 'name': 'transcript_fusion', 'dis...   \n",
       "1     [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "2     [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "3     [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "4     [{'id': 120, 'name': 'transcript_fusion', 'dis...   \n",
       "...                                                 ...   \n",
       "3051                                                 []   \n",
       "3052                                                 []   \n",
       "3053                                                 []   \n",
       "3054                                                 []   \n",
       "3055                                                 []   \n",
       "\n",
       "      civic_actionability_score  flagged                updated_at  \\\n",
       "0                         255.5    False  2020-02-08T03:53:21.874Z   \n",
       "1                         146.0    False  2020-08-29T00:06:08.064Z   \n",
       "2                          83.0    False  2021-06-08T23:03:08.149Z   \n",
       "3                          33.5    False  2021-11-12T18:53:34.889Z   \n",
       "4                          38.0    False  2020-08-29T19:17:30.276Z   \n",
       "...                         ...      ...                       ...   \n",
       "3051                        0.0    False  2022-01-07T06:56:37.814Z   \n",
       "3052                        0.0    False  2022-01-07T06:57:07.685Z   \n",
       "3053                        0.0    False  2022-01-07T06:58:48.728Z   \n",
       "3054                        0.0    False  2022-01-07T06:59:38.140Z   \n",
       "3055                        0.0    False  2022-01-07T20:57:08.903Z   \n",
       "\n",
       "                                            coordinates  \n",
       "0     {'chromosome': '22', 'start': 23522397, 'stop'...  \n",
       "1     {'chromosome': '9', 'start': 133748283, 'stop'...  \n",
       "2     {'chromosome': '9', 'start': 133738363, 'stop'...  \n",
       "3     {'chromosome': '14', 'start': 105246551, 'stop...  \n",
       "4     {'chromosome': '2', 'start': 42396490, 'stop':...  \n",
       "...                                                 ...  \n",
       "3051  {'chromosome': None, 'start': None, 'stop': No...  \n",
       "3052  {'chromosome': None, 'start': None, 'stop': No...  \n",
       "3053  {'chromosome': None, 'start': None, 'stop': No...  \n",
       "3054  {'chromosome': None, 'start': None, 'stop': No...  \n",
       "3055  {'chromosome': None, 'start': None, 'stop': No...  \n",
       "\n",
       "[3056 rows x 12 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "civic_variants = apit.Endpoint(url=\"https://civicdb.org/api/variants?count=3056\")\n",
    "df_civic_variants = civic_variants.data_as_pandas(\"records\")\n",
    "df_civic_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "### Load Evidence Items\n",
    "\n",
    "##### Load all records"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>disease</th>\n",
       "      <th>drugs</th>\n",
       "      <th>rating</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>status</th>\n",
       "      <th>open_change_count</th>\n",
       "      <th>type</th>\n",
       "      <th>source</th>\n",
       "      <th>flagged</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>updated_at</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>EID1</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>{'id': 3034, 'name': 'Lymphoid Leukemia', 'dis...</td>\n",
       "      <td>[]</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 51, 'name': 'The JAK2V617F activating m...</td>\n",
       "      <td>False</td>\n",
       "      <td>64</td>\n",
       "      <td>[]</td>\n",
       "      <td>2020-11-20T16:18:21.858Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>EID2</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>[]</td>\n",
       "      <td>3.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 52, 'name': 'A great majority of GISTs ...</td>\n",
       "      <td>False</td>\n",
       "      <td>99</td>\n",
       "      <td>[]</td>\n",
       "      <td>2015-06-21T16:49:38.876Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>EID3</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>{'id': 3, 'name': 'Acute Myeloid Leukemia', 'd...</td>\n",
       "      <td>[]</td>\n",
       "      <td>2.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 53, 'name': 'Do AML patients with DNMT3...</td>\n",
       "      <td>False</td>\n",
       "      <td>32</td>\n",
       "      <td>[]</td>\n",
       "      <td>2016-03-26T22:10:59.700Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>EID4</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>{'id': 3, 'name': 'Acute Myeloid Leukemia', 'd...</td>\n",
       "      <td>[]</td>\n",
       "      <td>3.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 54, 'name': 'Mutant DNMT3A: a marker of...</td>\n",
       "      <td>False</td>\n",
       "      <td>32</td>\n",
       "      <td>[]</td>\n",
       "      <td>2016-03-16T22:09:27.720Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>EID5</td>\n",
       "      <td>JAK2 V617F is associated with myeloid malignan...</td>\n",
       "      <td>{'id': 4, 'name': 'Chronic Myeloid Leukemia', ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 51, 'name': 'The JAK2V617F activating m...</td>\n",
       "      <td>False</td>\n",
       "      <td>64</td>\n",
       "      <td>[]</td>\n",
       "      <td>2015-06-21T16:49:39.053Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8574</th>\n",
       "      <td>9911</td>\n",
       "      <td>EID9911</td>\n",
       "      <td>A cohort of 141 patients with hemangioblastoma...</td>\n",
       "      <td>{'id': 2198, 'name': 'Von Hippel-Lindau Diseas...</td>\n",
       "      <td>[]</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>Predisposing</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>Rare Germline</td>\n",
       "      <td>None</td>\n",
       "      <td>submitted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 2324, 'name': 'The impact of molecular ...</td>\n",
       "      <td>False</td>\n",
       "      <td>2156</td>\n",
       "      <td>[{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...</td>\n",
       "      <td>2021-12-02T05:59:59.336Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8575</th>\n",
       "      <td>9912</td>\n",
       "      <td>EID9912</td>\n",
       "      <td>A cohort of 141 patients with hemangioblastoma...</td>\n",
       "      <td>{'id': 2198, 'name': 'Von Hippel-Lindau Diseas...</td>\n",
       "      <td>[]</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>Predisposing</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>Rare Germline</td>\n",
       "      <td>None</td>\n",
       "      <td>submitted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 2324, 'name': 'The impact of molecular ...</td>\n",
       "      <td>False</td>\n",
       "      <td>2114</td>\n",
       "      <td>[{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...</td>\n",
       "      <td>2021-12-02T06:00:47.234Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8576</th>\n",
       "      <td>9913</td>\n",
       "      <td>EID9913</td>\n",
       "      <td>A cohort of 141 patients with hemangioblastoma...</td>\n",
       "      <td>{'id': 2198, 'name': 'Von Hippel-Lindau Diseas...</td>\n",
       "      <td>[]</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>Predisposing</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>Rare Germline</td>\n",
       "      <td>None</td>\n",
       "      <td>submitted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 2324, 'name': 'The impact of molecular ...</td>\n",
       "      <td>False</td>\n",
       "      <td>2000</td>\n",
       "      <td>[{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...</td>\n",
       "      <td>2021-12-02T06:01:24.508Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8577</th>\n",
       "      <td>9914</td>\n",
       "      <td>EID9914</td>\n",
       "      <td>A cohort of 141 patients with hemangioblastoma...</td>\n",
       "      <td>{'id': 2198, 'name': 'Von Hippel-Lindau Diseas...</td>\n",
       "      <td>[]</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>Predisposing</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>Rare Germline</td>\n",
       "      <td>None</td>\n",
       "      <td>submitted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 2324, 'name': 'The impact of molecular ...</td>\n",
       "      <td>False</td>\n",
       "      <td>3634</td>\n",
       "      <td>[{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...</td>\n",
       "      <td>2021-12-02T06:02:24.472Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8578</th>\n",
       "      <td>9915</td>\n",
       "      <td>EID9915</td>\n",
       "      <td>A cohort of 141 patients with hemangioblastoma...</td>\n",
       "      <td>{'id': 2198, 'name': 'Von Hippel-Lindau Diseas...</td>\n",
       "      <td>[]</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>Predisposing</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>Rare Germline</td>\n",
       "      <td>None</td>\n",
       "      <td>submitted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 2324, 'name': 'The impact of molecular ...</td>\n",
       "      <td>False</td>\n",
       "      <td>1996</td>\n",
       "      <td>[{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...</td>\n",
       "      <td>2021-12-02T06:03:19.312Z</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8579 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        id     name                                        description  \\\n",
       "0        1     EID1  JAK2 V617F is not associated with lymphoid leu...   \n",
       "1        2     EID2  GIST tumors harboring PDGFRA D842V mutation ar...   \n",
       "2        3     EID3  DNMT3A R882 mutations occur most often in de n...   \n",
       "3        4     EID4  Young AML patients (<60 years old) with DNMT3A...   \n",
       "4        5     EID5  JAK2 V617F is associated with myeloid malignan...   \n",
       "...    ...      ...                                                ...   \n",
       "8574  9911  EID9911  A cohort of 141 patients with hemangioblastoma...   \n",
       "8575  9912  EID9912  A cohort of 141 patients with hemangioblastoma...   \n",
       "8576  9913  EID9913  A cohort of 141 patients with hemangioblastoma...   \n",
       "8577  9914  EID9914  A cohort of 141 patients with hemangioblastoma...   \n",
       "8578  9915  EID9915  A cohort of 141 patients with hemangioblastoma...   \n",
       "\n",
       "                                                disease drugs  rating  \\\n",
       "0     {'id': 3034, 'name': 'Lymphoid Leukemia', 'dis...    []     4.0   \n",
       "1     {'id': 2, 'name': 'Gastrointestinal Stromal Tu...    []     3.0   \n",
       "2     {'id': 3, 'name': 'Acute Myeloid Leukemia', 'd...    []     2.0   \n",
       "3     {'id': 3, 'name': 'Acute Myeloid Leukemia', 'd...    []     3.0   \n",
       "4     {'id': 4, 'name': 'Chronic Myeloid Leukemia', ...    []     4.0   \n",
       "...                                                 ...   ...     ...   \n",
       "8574  {'id': 2198, 'name': 'Von Hippel-Lindau Diseas...    []     2.0   \n",
       "8575  {'id': 2198, 'name': 'Von Hippel-Lindau Diseas...    []     2.0   \n",
       "8576  {'id': 2198, 'name': 'Von Hippel-Lindau Diseas...    []     2.0   \n",
       "8577  {'id': 2198, 'name': 'Von Hippel-Lindau Diseas...    []     2.0   \n",
       "8578  {'id': 2198, 'name': 'Von Hippel-Lindau Diseas...    []     2.0   \n",
       "\n",
       "     evidence_level evidence_type clinical_significance evidence_direction  \\\n",
       "0                 B    Diagnostic              Negative           Supports   \n",
       "1                 B    Diagnostic              Negative           Supports   \n",
       "2                 B    Diagnostic              Positive           Supports   \n",
       "3                 B    Diagnostic              Positive           Supports   \n",
       "4                 B    Diagnostic              Positive           Supports   \n",
       "...             ...           ...                   ...                ...   \n",
       "8574              C  Predisposing                   N/A                N/A   \n",
       "8575              C  Predisposing                   N/A                N/A   \n",
       "8576              C  Predisposing                   N/A                N/A   \n",
       "8577              C  Predisposing                   N/A                N/A   \n",
       "8578              C  Predisposing                   N/A                N/A   \n",
       "\n",
       "     variant_origin drug_interaction_type     status  open_change_count  \\\n",
       "0           Somatic                  None   accepted                  0   \n",
       "1           Somatic                  None   accepted                  0   \n",
       "2           Somatic                  None   accepted                  0   \n",
       "3           Somatic                  None   accepted                  0   \n",
       "4           Somatic                  None   accepted                  0   \n",
       "...             ...                   ...        ...                ...   \n",
       "8574  Rare Germline                  None  submitted                  0   \n",
       "8575  Rare Germline                  None  submitted                  0   \n",
       "8576  Rare Germline                  None  submitted                  0   \n",
       "8577  Rare Germline                  None  submitted                  0   \n",
       "8578  Rare Germline                  None  submitted                  0   \n",
       "\n",
       "          type                                             source  flagged  \\\n",
       "0     evidence  {'id': 51, 'name': 'The JAK2V617F activating m...    False   \n",
       "1     evidence  {'id': 52, 'name': 'A great majority of GISTs ...    False   \n",
       "2     evidence  {'id': 53, 'name': 'Do AML patients with DNMT3...    False   \n",
       "3     evidence  {'id': 54, 'name': 'Mutant DNMT3A: a marker of...    False   \n",
       "4     evidence  {'id': 51, 'name': 'The JAK2V617F activating m...    False   \n",
       "...        ...                                                ...      ...   \n",
       "8574  evidence  {'id': 2324, 'name': 'The impact of molecular ...    False   \n",
       "8575  evidence  {'id': 2324, 'name': 'The impact of molecular ...    False   \n",
       "8576  evidence  {'id': 2324, 'name': 'The impact of molecular ...    False   \n",
       "8577  evidence  {'id': 2324, 'name': 'The impact of molecular ...    False   \n",
       "8578  evidence  {'id': 2324, 'name': 'The impact of molecular ...    False   \n",
       "\n",
       "      variant_id                                         phenotypes  \\\n",
       "0             64                                                 []   \n",
       "1             99                                                 []   \n",
       "2             32                                                 []   \n",
       "3             32                                                 []   \n",
       "4             64                                                 []   \n",
       "...          ...                                                ...   \n",
       "8574        2156  [{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...   \n",
       "8575        2114  [{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...   \n",
       "8576        2000  [{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...   \n",
       "8577        3634  [{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...   \n",
       "8578        1996  [{'id': 7467, 'hpo_id': 'HP:0010797', 'hpo_cla...   \n",
       "\n",
       "                    updated_at  \n",
       "0     2020-11-20T16:18:21.858Z  \n",
       "1     2015-06-21T16:49:38.876Z  \n",
       "2     2016-03-26T22:10:59.700Z  \n",
       "3     2016-03-16T22:09:27.720Z  \n",
       "4     2015-06-21T16:49:39.053Z  \n",
       "...                        ...  \n",
       "8574  2021-12-02T05:59:59.336Z  \n",
       "8575  2021-12-02T06:00:47.234Z  \n",
       "8576  2021-12-02T06:01:24.508Z  \n",
       "8577  2021-12-02T06:02:24.472Z  \n",
       "8578  2021-12-02T06:03:19.312Z  \n",
       "\n",
       "[8579 rows x 20 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "civic_evidence = apit.Endpoint(url=\"https://civicdb.org/api/evidence_items?count=8579\")\n",
    "df_civic_evidence = civic_evidence.data_as_pandas(\"records\")\n",
    "df_civic_evidence"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "##### Filter evidence items for records with \"Predictive\" evidence type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>disease</th>\n",
       "      <th>drugs</th>\n",
       "      <th>rating</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>status</th>\n",
       "      <th>open_change_count</th>\n",
       "      <th>type</th>\n",
       "      <th>source</th>\n",
       "      <th>flagged</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>updated_at</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>12</td>\n",
       "      <td>EID12</td>\n",
       "      <td>A375 cells expressing MAP2K1 P124S mutation co...</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>[{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>D</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 58, 'name': 'MEK1 mutations confer resi...</td>\n",
       "      <td>False</td>\n",
       "      <td>82</td>\n",
       "      <td>[]</td>\n",
       "      <td>2016-08-19T18:23:15.824Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>13</td>\n",
       "      <td>EID13</td>\n",
       "      <td>A375 cells expressing MAP2K1 Q56P mutation con...</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>[{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>D</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 58, 'name': 'MEK1 mutations confer resi...</td>\n",
       "      <td>False</td>\n",
       "      <td>83</td>\n",
       "      <td>[]</td>\n",
       "      <td>2016-11-07T18:43:41.247Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>14</td>\n",
       "      <td>EID14</td>\n",
       "      <td>Vemurafenib resistance is associated with gain...</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>[{'id': 4, 'name': 'Vemurafenib', 'ncit_id': '...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 60, 'name': 'Pharmacodynamic effects an...</td>\n",
       "      <td>False</td>\n",
       "      <td>94</td>\n",
       "      <td>[]</td>\n",
       "      <td>2015-06-21T16:49:39.376Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>15</td>\n",
       "      <td>EID15</td>\n",
       "      <td>GIST cancer with D842V mutation is resistant t...</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>[{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 61, 'name': 'PDGFRA mutations in gastro...</td>\n",
       "      <td>False</td>\n",
       "      <td>99</td>\n",
       "      <td>[]</td>\n",
       "      <td>2015-06-21T16:49:39.416Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>16</td>\n",
       "      <td>EID16</td>\n",
       "      <td>While cancer with PDGFRA V561D mutation is kno...</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>[{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>None</td>\n",
       "      <td>accepted</td>\n",
       "      <td>0</td>\n",
       "      <td>evidence</td>\n",
       "      <td>{'id': 62, 'name': 'Molecular correlates of im...</td>\n",
       "      <td>False</td>\n",
       "      <td>99</td>\n",
       "      <td>[]</td>\n",
       "      <td>2016-12-01T15:26:38.682Z</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    id   name                                        description  \\\n",
       "9   12  EID12  A375 cells expressing MAP2K1 P124S mutation co...   \n",
       "10  13  EID13  A375 cells expressing MAP2K1 Q56P mutation con...   \n",
       "11  14  EID14  Vemurafenib resistance is associated with gain...   \n",
       "12  15  EID15  GIST cancer with D842V mutation is resistant t...   \n",
       "13  16  EID16  While cancer with PDGFRA V561D mutation is kno...   \n",
       "\n",
       "                                              disease  \\\n",
       "9   {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "10  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "11  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "12  {'id': 2, 'name': 'Gastrointestinal Stromal Tu...   \n",
       "13  {'id': 2, 'name': 'Gastrointestinal Stromal Tu...   \n",
       "\n",
       "                                                drugs  rating evidence_level  \\\n",
       "9   [{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...     3.0              D   \n",
       "10  [{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...     3.0              D   \n",
       "11  [{'id': 4, 'name': 'Vemurafenib', 'ncit_id': '...     4.0              B   \n",
       "12  [{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...     4.0              B   \n",
       "13  [{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...     4.0              B   \n",
       "\n",
       "   evidence_type clinical_significance evidence_direction variant_origin  \\\n",
       "9     Predictive            Resistance           Supports        Somatic   \n",
       "10    Predictive            Resistance           Supports        Somatic   \n",
       "11    Predictive            Resistance           Supports        Somatic   \n",
       "12    Predictive            Resistance           Supports        Somatic   \n",
       "13    Predictive            Resistance           Supports        Somatic   \n",
       "\n",
       "   drug_interaction_type    status  open_change_count      type  \\\n",
       "9                   None  accepted                  0  evidence   \n",
       "10                  None  accepted                  0  evidence   \n",
       "11                  None  accepted                  0  evidence   \n",
       "12                  None  accepted                  0  evidence   \n",
       "13                  None  accepted                  0  evidence   \n",
       "\n",
       "                                               source  flagged  variant_id  \\\n",
       "9   {'id': 58, 'name': 'MEK1 mutations confer resi...    False          82   \n",
       "10  {'id': 58, 'name': 'MEK1 mutations confer resi...    False          83   \n",
       "11  {'id': 60, 'name': 'Pharmacodynamic effects an...    False          94   \n",
       "12  {'id': 61, 'name': 'PDGFRA mutations in gastro...    False          99   \n",
       "13  {'id': 62, 'name': 'Molecular correlates of im...    False          99   \n",
       "\n",
       "   phenotypes                updated_at  \n",
       "9          []  2016-08-19T18:23:15.824Z  \n",
       "10         []  2016-11-07T18:43:41.247Z  \n",
       "11         []  2015-06-21T16:49:39.376Z  \n",
       "12         []  2015-06-21T16:49:39.416Z  \n",
       "13         []  2016-12-01T15:26:38.682Z  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_civic_evidence_filtered = df_civic_evidence[df_civic_evidence[\"evidence_type\"] == \"Predictive\"]\n",
    "df_civic_evidence_filtered = df_civic_evidence_filtered[df_civic_evidence_filtered[\"evidence_direction\"] == \"Supports\"]\n",
    "df_civic_evidence_filtered = df_civic_evidence_filtered[df_civic_evidence_filtered[\"clinical_significance\"] != \"N/A\"]\n",
    "df_civic_evidence_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "raw",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "##### Left-join the variant table to the filtered evidence item table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_x</th>\n",
       "      <th>name_x</th>\n",
       "      <th>description_x</th>\n",
       "      <th>disease</th>\n",
       "      <th>drugs</th>\n",
       "      <th>rating</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>...</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>name_y</th>\n",
       "      <th>description_y</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>type_y</th>\n",
       "      <th>variant_types</th>\n",
       "      <th>civic_actionability_score</th>\n",
       "      <th>flagged_y</th>\n",
       "      <th>updated_at_y</th>\n",
       "      <th>coordinates</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>EID12</td>\n",
       "      <td>A375 cells expressing MAP2K1 P124S mutation co...</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>[{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>D</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>P124S</td>\n",
       "      <td>MAP2K1 P124S is a recurrent mutation in melano...</td>\n",
       "      <td>31.0</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-11-30T14:43:05.546Z</td>\n",
       "      <td>{'chromosome': '15', 'start': 66729162, 'stop'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13</td>\n",
       "      <td>EID13</td>\n",
       "      <td>A375 cells expressing MAP2K1 Q56P mutation con...</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>[{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>D</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>MAP2K1 Q56P is a recurrent mutation in melanom...</td>\n",
       "      <td>31.0</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2018-11-30T14:43:05.683Z</td>\n",
       "      <td>{'chromosome': '15', 'start': 66727451, 'stop'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>EID14</td>\n",
       "      <td>Vemurafenib resistance is associated with gain...</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>[{'id': 4, 'name': 'Vemurafenib', 'ncit_id': '...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Q61</td>\n",
       "      <td>NRAS Q61 mutations have been found in multiple...</td>\n",
       "      <td>36.0</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>False</td>\n",
       "      <td>2020-11-25T00:09:17.599Z</td>\n",
       "      <td>{'chromosome': '1', 'start': 115256528, 'stop'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>EID15</td>\n",
       "      <td>GIST cancer with D842V mutation is resistant t...</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>[{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>D842V</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>38.0</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>100.5</td>\n",
       "      <td>False</td>\n",
       "      <td>2019-05-03T23:38:41.672Z</td>\n",
       "      <td>{'chromosome': '4', 'start': 55152093, 'stop':...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16</td>\n",
       "      <td>EID16</td>\n",
       "      <td>While cancer with PDGFRA V561D mutation is kno...</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>[{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>B</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>D842V</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>38.0</td>\n",
       "      <td>variant</td>\n",
       "      <td>[{'id': 47, 'name': 'missense_variant', 'displ...</td>\n",
       "      <td>100.5</td>\n",
       "      <td>False</td>\n",
       "      <td>2019-05-03T23:38:41.672Z</td>\n",
       "      <td>{'chromosome': '4', 'start': 55152093, 'stop':...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   id_x name_x                                      description_x  \\\n",
       "0    12  EID12  A375 cells expressing MAP2K1 P124S mutation co...   \n",
       "1    13  EID13  A375 cells expressing MAP2K1 Q56P mutation con...   \n",
       "2    14  EID14  Vemurafenib resistance is associated with gain...   \n",
       "3    15  EID15  GIST cancer with D842V mutation is resistant t...   \n",
       "4    16  EID16  While cancer with PDGFRA V561D mutation is kno...   \n",
       "\n",
       "                                             disease  \\\n",
       "0  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "1  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "2  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "3  {'id': 2, 'name': 'Gastrointestinal Stromal Tu...   \n",
       "4  {'id': 2, 'name': 'Gastrointestinal Stromal Tu...   \n",
       "\n",
       "                                               drugs  rating evidence_level  \\\n",
       "0  [{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...     3.0              D   \n",
       "1  [{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...     3.0              D   \n",
       "2  [{'id': 4, 'name': 'Vemurafenib', 'ncit_id': '...     4.0              B   \n",
       "3  [{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...     4.0              B   \n",
       "4  [{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...     4.0              B   \n",
       "\n",
       "  evidence_type clinical_significance evidence_direction  ... entrez_id  \\\n",
       "0    Predictive            Resistance           Supports  ...    5604.0   \n",
       "1    Predictive            Resistance           Supports  ...    5604.0   \n",
       "2    Predictive            Resistance           Supports  ...    4893.0   \n",
       "3    Predictive            Resistance           Supports  ...    5156.0   \n",
       "4    Predictive            Resistance           Supports  ...    5156.0   \n",
       "\n",
       "  name_y                                      description_y  gene_id   type_y  \\\n",
       "0  P124S  MAP2K1 P124S is a recurrent mutation in melano...     31.0  variant   \n",
       "1   Q56P  MAP2K1 Q56P is a recurrent mutation in melanom...     31.0  variant   \n",
       "2    Q61  NRAS Q61 mutations have been found in multiple...     36.0  variant   \n",
       "3  D842V  PDGFRA D842 mutations are characterized broadl...     38.0  variant   \n",
       "4  D842V  PDGFRA D842 mutations are characterized broadl...     38.0  variant   \n",
       "\n",
       "                                       variant_types  \\\n",
       "0  [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "1  [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "2  [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "3  [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "4  [{'id': 47, 'name': 'missense_variant', 'displ...   \n",
       "\n",
       "   civic_actionability_score  flagged_y              updated_at_y  \\\n",
       "0                        3.0      False  2018-11-30T14:43:05.546Z   \n",
       "1                        3.0      False  2018-11-30T14:43:05.683Z   \n",
       "2                      100.0      False  2020-11-25T00:09:17.599Z   \n",
       "3                      100.5      False  2019-05-03T23:38:41.672Z   \n",
       "4                      100.5      False  2019-05-03T23:38:41.672Z   \n",
       "\n",
       "                                         coordinates  \n",
       "0  {'chromosome': '15', 'start': 66729162, 'stop'...  \n",
       "1  {'chromosome': '15', 'start': 66727451, 'stop'...  \n",
       "2  {'chromosome': '1', 'start': 115256528, 'stop'...  \n",
       "3  {'chromosome': '4', 'start': 55152093, 'stop':...  \n",
       "4  {'chromosome': '4', 'start': 55152093, 'stop':...  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variant_evidence = df_civic_evidence_filtered.merge(df_civic_variants, left_on=\"variant_id\", right_on=\"id\", how=\"left\")\n",
    "df_variant_evidence.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "##### Isolate the columns of interest and obtain the drug names associated with each variant and evidence item"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name_y</th>\n",
       "      <th>drugs</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>disease</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P124S</td>\n",
       "      <td>[{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>5604.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q56P</td>\n",
       "      <td>[{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>5604.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q61</td>\n",
       "      <td>[{'id': 4, 'name': 'Vemurafenib', 'ncit_id': '...</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>{'id': 7, 'name': 'Melanoma', 'display_name': ...</td>\n",
       "      <td>4893.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D842V</td>\n",
       "      <td>[{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>5156.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D842V</td>\n",
       "      <td>[{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>{'id': 2, 'name': 'Gastrointestinal Stromal Tu...</td>\n",
       "      <td>5156.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  name_y                                              drugs evidence_level  \\\n",
       "0  P124S  [{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...              D   \n",
       "1   Q56P  [{'id': 63, 'name': 'Selumetinib', 'ncit_id': ...              D   \n",
       "2    Q61  [{'id': 4, 'name': 'Vemurafenib', 'ncit_id': '...              B   \n",
       "3  D842V  [{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...              B   \n",
       "4  D842V  [{'id': 5, 'name': 'Imatinib', 'ncit_id': 'C62...              B   \n",
       "\n",
       "  clinical_significance                                            disease  \\\n",
       "0            Resistance  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "1            Resistance  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "2            Resistance  {'id': 7, 'name': 'Melanoma', 'display_name': ...   \n",
       "3            Resistance  {'id': 2, 'name': 'Gastrointestinal Stromal Tu...   \n",
       "4            Resistance  {'id': 2, 'name': 'Gastrointestinal Stromal Tu...   \n",
       "\n",
       "   entrez_id  \n",
       "0     5604.0  \n",
       "1     5604.0  \n",
       "2     4893.0  \n",
       "3     5156.0  \n",
       "4     5156.0  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_of_interest = [\"name_y\", \"drugs\", \"evidence_level\", \"clinical_significance\", \"disease\", \"entrez_id\"]\n",
    "df_variant_evidence_filtered = df_variant_evidence.loc[:, columns_of_interest]\n",
    "df_variant_evidence_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "for i in range(df_variant_evidence_filtered.shape[0]):\n",
    "    drug_list = [drug[\"name\"] for drug in df_variant_evidence_filtered.loc[i, \"drugs\"]]\n",
    "    therapy_regimen = \"+\".join(drug_list)\n",
    "    df_variant_evidence_filtered.loc[i, \"therapy_regimen\"] = therapy_regimen\n",
    "    df_variant_evidence_filtered.loc[i, \"disease\"] = df_variant_evidence_filtered.loc[i, \"disease\"][\"name\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Obtain gene names from `entrez_id`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name_y</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>disease</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>therapy_regimen</th>\n",
       "      <th>Gene_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P124S</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q56P</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q61</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Vemurafenib</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D842V</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D842V</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  name_y evidence_level clinical_significance                         disease  \\\n",
       "0  P124S              D            Resistance                        Melanoma   \n",
       "1   Q56P              D            Resistance                        Melanoma   \n",
       "2    Q61              B            Resistance                        Melanoma   \n",
       "3  D842V              B            Resistance  Gastrointestinal Stromal Tumor   \n",
       "4  D842V              B            Resistance  Gastrointestinal Stromal Tumor   \n",
       "\n",
       "   entrez_id therapy_regimen Gene_Symbol  \n",
       "0     5604.0     Selumetinib      MAP2K1  \n",
       "1     5604.0     Selumetinib      MAP2K1  \n",
       "2     4893.0     Vemurafenib        NRAS  \n",
       "3     5156.0        Imatinib      PDGFRA  \n",
       "4     5156.0        Imatinib      PDGFRA  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list = pd.read_csv(\"CancerGeneList.tsv\", sep=\"\\t\", header=0, usecols=[0, 1])\n",
    "df_variant_evidence_filtered = df_variant_evidence_filtered.merge(gene_list, left_on=\"entrez_id\", right_on=\"Entrez_Id\", how=\"left\")\n",
    "df_variant_evidence_filtered = df_variant_evidence_filtered.drop(columns=[\"Entrez_Id\", \"drugs\"])\n",
    "df_variant_evidence_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>EvidenceLevel</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>Disease</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>TherapyRegimen</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P124S</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q56P</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q61</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Vemurafenib</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D842V</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D842V</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S214C</td>\n",
       "      <td>C</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Lung Non-small Cell Carcinoma</td>\n",
       "      <td>369.0</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>ARAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>R882</td>\n",
       "      <td>B</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>1788.0</td>\n",
       "      <td>Idarubicin</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>V617F</td>\n",
       "      <td>B</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Polycythemia Vera</td>\n",
       "      <td>3717.0</td>\n",
       "      <td></td>\n",
       "      <td>JAK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>V617F</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Polycythemia Vera</td>\n",
       "      <td>3717.0</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>JAK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>G13D</td>\n",
       "      <td>C</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Tanespimycin</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Q61L</td>\n",
       "      <td>C</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Temozolomide</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Q61R</td>\n",
       "      <td>C</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Temozolomide</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>F1174L</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Inflammatory Myofibroblastic Tumor</td>\n",
       "      <td>238.0</td>\n",
       "      <td>Crizotinib</td>\n",
       "      <td>ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>F1174L</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Lung Non-small Cell Carcinoma</td>\n",
       "      <td>238.0</td>\n",
       "      <td>Crizotinib</td>\n",
       "      <td>ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>L576P</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>3815.0</td>\n",
       "      <td>Nilotinib+Imatinib+Sorafenib</td>\n",
       "      <td>KIT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>G12/G13</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Lung Non-small Cell Carcinoma</td>\n",
       "      <td>3845.0</td>\n",
       "      <td>Gefitinib+Erlotinib</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Q61</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Cetuximab+Chemotherapy</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>F1174L</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Neuroblastoma</td>\n",
       "      <td>238.0</td>\n",
       "      <td>Alectinib</td>\n",
       "      <td>ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>F1174L</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Neuroblastoma</td>\n",
       "      <td>238.0</td>\n",
       "      <td>Crizotinib</td>\n",
       "      <td>ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>R1275Q</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Neuroblastoma</td>\n",
       "      <td>238.0</td>\n",
       "      <td>TAE684</td>\n",
       "      <td>ALK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant EvidenceLevel  ClinicalSignificance  \\\n",
       "0     P124S             D            Resistance   \n",
       "1      Q56P             D            Resistance   \n",
       "2       Q61             B            Resistance   \n",
       "3     D842V             B            Resistance   \n",
       "4     D842V             B            Resistance   \n",
       "5     S214C             C  Sensitivity/Response   \n",
       "6      R882             B  Sensitivity/Response   \n",
       "7     V617F             B  Sensitivity/Response   \n",
       "8     V617F             D  Sensitivity/Response   \n",
       "9      G13D             C  Sensitivity/Response   \n",
       "10     Q61L             C  Sensitivity/Response   \n",
       "11     Q61R             C  Sensitivity/Response   \n",
       "12   F1174L             D            Resistance   \n",
       "13   F1174L             D            Resistance   \n",
       "14    L576P             D            Resistance   \n",
       "15  G12/G13             B            Resistance   \n",
       "16      Q61             B            Resistance   \n",
       "17   F1174L             D  Sensitivity/Response   \n",
       "18   F1174L             D  Sensitivity/Response   \n",
       "19   R1275Q             D  Sensitivity/Response   \n",
       "\n",
       "                               Disease  entrez_id  \\\n",
       "0                             Melanoma     5604.0   \n",
       "1                             Melanoma     5604.0   \n",
       "2                             Melanoma     4893.0   \n",
       "3       Gastrointestinal Stromal Tumor     5156.0   \n",
       "4       Gastrointestinal Stromal Tumor     5156.0   \n",
       "5        Lung Non-small Cell Carcinoma      369.0   \n",
       "6               Acute Myeloid Leukemia     1788.0   \n",
       "7                    Polycythemia Vera     3717.0   \n",
       "8                    Polycythemia Vera     3717.0   \n",
       "9                             Melanoma     4893.0   \n",
       "10                            Melanoma     4893.0   \n",
       "11                            Melanoma     4893.0   \n",
       "12  Inflammatory Myofibroblastic Tumor      238.0   \n",
       "13       Lung Non-small Cell Carcinoma      238.0   \n",
       "14                            Melanoma     3815.0   \n",
       "15       Lung Non-small Cell Carcinoma     3845.0   \n",
       "16                   Colorectal Cancer     4893.0   \n",
       "17                       Neuroblastoma      238.0   \n",
       "18                       Neuroblastoma      238.0   \n",
       "19                       Neuroblastoma      238.0   \n",
       "\n",
       "                  TherapyRegimen    Gene  \n",
       "0                    Selumetinib  MAP2K1  \n",
       "1                    Selumetinib  MAP2K1  \n",
       "2                    Vemurafenib    NRAS  \n",
       "3                       Imatinib  PDGFRA  \n",
       "4                       Imatinib  PDGFRA  \n",
       "5                      Sorafenib    ARAF  \n",
       "6                     Idarubicin  DNMT3A  \n",
       "7                                   JAK2  \n",
       "8                     Fedratinib    JAK2  \n",
       "9                   Tanespimycin    NRAS  \n",
       "10                  Temozolomide    NRAS  \n",
       "11                  Temozolomide    NRAS  \n",
       "12                    Crizotinib     ALK  \n",
       "13                    Crizotinib     ALK  \n",
       "14  Nilotinib+Imatinib+Sorafenib     KIT  \n",
       "15           Gefitinib+Erlotinib    KRAS  \n",
       "16        Cetuximab+Chemotherapy    NRAS  \n",
       "17                     Alectinib     ALK  \n",
       "18                    Crizotinib     ALK  \n",
       "19                        TAE684     ALK  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "column_mappings = {\n",
    "    \"name_y\": \"variant\",\n",
    "    \"therapy_regimen\": \"TherapyRegimen\",\n",
    "    \"evidence_level\": \"EvidenceLevel\",\n",
    "    \"clinical_significance\": \"ClinicalSignificance\",\n",
    "    \"disease\": \"Disease\",\n",
    "    \"Gene_Symbol\": \"Gene\"\n",
    "}\n",
    "df_variant_evidence_filtered.rename(columns=column_mappings, inplace=True)\n",
    "df_variant_evidence_filtered.head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "##### Safe Table to CSV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "df_variant_evidence_filtered.to_csv(\"CIViC_variant_evidence.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>EvidenceLevel</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>Disease</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>TherapyRegimen</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P124S</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q56P</td>\n",
       "      <td>D</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>5604.0</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q61</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>4893.0</td>\n",
       "      <td>Vemurafenib</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D842V</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D842V</td>\n",
       "      <td>B</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>5156.0</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4014</th>\n",
       "      <td>High expression</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Stomach Cancer</td>\n",
       "      <td>4921.0</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>DDR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4015</th>\n",
       "      <td>Overexpression</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Head And Neck Carcinoma</td>\n",
       "      <td>4921.0</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>DDR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4016</th>\n",
       "      <td>Overexpression</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Esophageal Cancer</td>\n",
       "      <td>9734.0</td>\n",
       "      <td>HDAC Inhibitor AR-42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4017</th>\n",
       "      <td>Overexpression</td>\n",
       "      <td>D</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Esophageal Cancer</td>\n",
       "      <td>10013.0</td>\n",
       "      <td>Trichostatin A+Vorinostat+Panobinostat</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4018</th>\n",
       "      <td>V600E and Amplification</td>\n",
       "      <td>B</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>Epithelial Ovarian Cancer</td>\n",
       "      <td>673.0</td>\n",
       "      <td>Bevacizumab+Atezolizumab</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4019 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      variant EvidenceLevel  ClinicalSignificance  \\\n",
       "0                       P124S             D            Resistance   \n",
       "1                        Q56P             D            Resistance   \n",
       "2                         Q61             B            Resistance   \n",
       "3                       D842V             B            Resistance   \n",
       "4                       D842V             B            Resistance   \n",
       "...                       ...           ...                   ...   \n",
       "4014          High expression             D  Sensitivity/Response   \n",
       "4015           Overexpression             D  Sensitivity/Response   \n",
       "4016           Overexpression             D  Sensitivity/Response   \n",
       "4017           Overexpression             D  Sensitivity/Response   \n",
       "4018  V600E and Amplification             B  Sensitivity/Response   \n",
       "\n",
       "                             Disease  entrez_id  \\\n",
       "0                           Melanoma     5604.0   \n",
       "1                           Melanoma     5604.0   \n",
       "2                           Melanoma     4893.0   \n",
       "3     Gastrointestinal Stromal Tumor     5156.0   \n",
       "4     Gastrointestinal Stromal Tumor     5156.0   \n",
       "...                              ...        ...   \n",
       "4014                  Stomach Cancer     4921.0   \n",
       "4015         Head And Neck Carcinoma     4921.0   \n",
       "4016               Esophageal Cancer     9734.0   \n",
       "4017               Esophageal Cancer    10013.0   \n",
       "4018       Epithelial Ovarian Cancer      673.0   \n",
       "\n",
       "                              TherapyRegimen    Gene  \n",
       "0                                Selumetinib  MAP2K1  \n",
       "1                                Selumetinib  MAP2K1  \n",
       "2                                Vemurafenib    NRAS  \n",
       "3                                   Imatinib  PDGFRA  \n",
       "4                                   Imatinib  PDGFRA  \n",
       "...                                      ...     ...  \n",
       "4014                               Dasatinib    DDR2  \n",
       "4015                               Dasatinib    DDR2  \n",
       "4016                    HDAC Inhibitor AR-42     NaN  \n",
       "4017  Trichostatin A+Vorinostat+Panobinostat     NaN  \n",
       "4018                Bevacizumab+Atezolizumab    BRAF  \n",
       "\n",
       "[4019 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variant_evidence_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
