---
title: "Prepare TCGA PANCAN Data"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(maftools)
library(tidyverse)
```


## SOURCE FILE INFO AND FILTERING CRITERIA
```{r filter_criteria}
RESTRICT_TO_CANCER_GENES = TRUE

# Somatic SNV/Indel -----------------------------------------------------------
MIN_NORMAL_WXS_DEPTH <- 20
MIN_TUMOR_WXS_DEPTH <- 50
MIN_TUMOR_ALT_READ_COUNT <- 5

# Somatic Copy Number ---------------------------------------------------------
xena_genecop_url <- "https://tcga-xena-hub.s3.us-east-1.amazonaws.com/download/TCGA.PANCAN.sampleMap%2FGistic2_CopyNumber_Gistic2_all_thresholded.by_genes.gz"
FILTER_LOW_LEVEL_COPY_GAINS <- TRUE

# Gene Fusion -----------------------------------------------------------------
MIN_SPANNING_READ_COUNT <- 3
MIN_JUNCTION_READ_COUNT <- 2

```


## ASSEMBLE CANCER GENE LIST
```{r cangenes}
cosmic_cangenes <- readr::read_tsv(
  "COSMIC_Census_allTue_Feb_22_00_32_56_2022.tsv",
  show_col_types = FALSE
)
oncokb_cangenes <- readr::read_tsv(
  "OncoKB_cancerGeneList.tsv",
  show_col_types = FALSE
)
tcga_dna_repair_genes <- readxl::read_xlsx(
  "TCGA_DNA_Repair_Genes_PMID_29617664.xlsx", skip = 3)

cangenes <- c(
  cosmic_cangenes$`Gene Symbol`,
  oncokb_cangenes$`Hugo Symbol`,
  tcga_dna_repair_genes$`Gene Symbol`
) %>%
  unique() %>%
  stringr::str_trim() %>%
  sort()

# readr::write_tsv(tibble(Gene_Symbol = cangenes), file = "CancerGeneList.tsv")
```


## LOAD TCGA PANCAN SOMATIC MUTATION DATA
```{r utils}
# https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/
# Tumor_Sample_Barcode: TCGA-OR-A5J1-01A-11D-A29I-10
# Participant_ID: TCGA-OR-A5J1
# Sample_ID: TCGA-OR-A5J1-01

barcode_to_participant <- function(x) {
  stringr::str_split(x, pattern = "-") %>%
    purrr::map_chr(function(parts) {
      paste0(parts[1:3], collapse = "-")
    })
}


barcode_to_sample <- function(x) {
  stringr::str_split(x, pattern = "-") %>%
    purrr::map_chr(function(parts) {
      paste0(parts[1:4], collapse = "-")
    }) %>%
    purrr::map_chr(function(s) {
      stringr::str_sub(s, 1, nchar(s) - 1)
    })
}
```


```{r maf_tbl}
tcga_cohorts = maftools::tcgaAvailable() %>%
  filter(Study_Abbreviation != "Unknown") 

# Construct: tibble data frame with each maf column entry containing TCGA study 
# (cancer type) specific (non-silent) mutation data in maf format.
tcga_maf = tcga_cohorts %>%
  select(Study_Abbreviation, Study_Name) %>%
  mutate(
    maf = purrr::map(Study_Abbreviation, function(cohort) {
      maftools::tcgaLoad(cohort, source = "MC3")@data %>% 
        as_tibble() %>%
        mutate_if(is.factor, as.character)
    })
  )

# Assert: each maf table has the same (ordered) columns.
expected_cols <- colnames(tcga_maf$maf[[1]])
stopifnot(
  all(
    purrr::map_lgl(tcga_maf$maf, function(df) {
      identical(colnames(df), expected_cols)
    })
  )
)

tcga_maf <- tcga_maf %>% tidyr::unnest(cols = "maf")
stopifnot(identical(tcga_maf$Study_Abbreviation, tcga_maf$Cohort))
tcga_maf <- tcga_maf %>% select(-Cohort)
```


## MUTATION TYPE SUMMARY STATISTICS (PRE-FILTERING)
```{r mut_stats_prefilter}
table(tcga_maf$Variant_Type)
table(tcga_maf$Variant_Classification)

  #   DEL     INS     ONP     SNP     TNP 
  # 99640   28494      81 2116712      22 
  # 
  #      Frame_Shift_Del        Frame_Shift_Ins           In_Frame_Del 
  #                86680                  27033                  10218 
  #         In_Frame_Ins      Missense_Mutation      Nonsense_Mutation 
  #                  892                1909060                 156170 
  #     Nonstop_Mutation            Splice_Site Translation_Start_Site 
  #                 2026                  50167                   2703 

```


## FILTER SOMATIC MUTATION TABLE
```{r mut_filters}
if (RESTRICT_TO_CANCER_GENES) {
  tcga_maf <- tcga_maf %>% filter(Hugo_Symbol %in% cangenes)
}

tcga_maf <- tcga_maf %>%
  filter(n_depth >= MIN_NORMAL_WXS_DEPTH) %>%
  filter(t_depth >= MIN_TUMOR_WXS_DEPTH)  %>%
  filter(t_alt_count >= MIN_TUMOR_ALT_READ_COUNT) %>%
  mutate(
    participant_id = barcode_to_participant(Tumor_Sample_Barcode),
    sample_id = barcode_to_sample(Tumor_Sample_Barcode)
  ) %>%
  select(participant_id, sample_id, Tumor_Sample_Barcode, everything())

```


## SAMPLE AND MUTATION TYPE SUMMARY STATISTICS (POST-FILTERING)
```{r mut_stats_prefilter}
# Samples per participant:
samples_per_participant <- tcga_maf %>%
  select(participant_id, sample_id) %>%
  unique() %>%
  group_by(participant_id) %>%
  summarize(n = n())
table(samples_per_participant$n)

# Barcodes per sample:
barcodes_per_sample <- tcga_maf %>%
  select(sample_id, Tumor_Sample_Barcode) %>%
  unique() %>%
  group_by(sample_id) %>%
  summarize(n = n())
table(barcodes_per_sample$n)

table(tcga_maf$Variant_Type)
table(tcga_maf$Variant_Classification)

#    DEL    INS    ONP    SNP    TNP 
#  10872   3617      4 137439      0 
# 
#        Frame_Shift_Del        Frame_Shift_Ins           In_Frame_Del 
#                   9387                   3375                   1152 
#           In_Frame_Ins      Missense_Mutation      Nonsense_Mutation 
#                    134                 121515                  12657 
#       Nonstop_Mutation            Splice_Site Translation_Start_Site 
#                     87                   3511                    114 
# 
```


--------------------------------------------------------------------------------

## LOAD TCGA PANCAN (GENE-LEVEL) SOMATIC COPY ALTERATION DATA
```{r load_copyalt}
system2(command = "wget", args    = xena_genecop_url)
  
tcga_mut_samples <- tcga_maf$sample_id %>% unique()

Sys.setenv(VROOM_CONNECTION_SIZE = "10000000 KiB")
tcga_cop <- readr::read_tsv(
  "TCGA.PANCAN.sampleMap%2FGistic2_CopyNumber_Gistic2_all_thresholded.by_genes.gz"
) %>%
  rename(gene = Sample) %>%
  filter(gene %in% cangenes) %>%
  tidyr::pivot_longer(cols = setdiff(colnames(.), "gene"), 
                      names_to = "sample_id", values_to = "copy_state") %>%
  filter(copy_state != 0) %>%
  filter(sample_id %in% tcga_mut_samples) %>%
  select(sample_id, gene, copy_state) %>%
  arrange(sample_id, gene)

if (FILTER_LOW_LEVEL_COPY_GAINS) {
  tcga_cop <- filter(tcga_cop, tcga_cop$copy_state != 1)
}


# SOURCE FILE DETAILS ----------------------------------------------------------
# https://xenabrowser.net/datapages/?dataset=TCGA.PANCAN.sampleMap%2FGistic2_CopyNumber_Gistic2_all_thresholded.by_genes&host=https%3A%2F%2Ftcga.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443

# {"longTitle":
#  "TCGA pan-cancer gene-level copy number (gistic2_thresholded)",
#  "cohort":"TCGA Pan-Cancer (PANCAN)",
# ":probeMap":"/probeMap/hugo_gencode_good_hg19_V24lift37_probemap",
#  "url":
#  "https://tcga-data.nci.nih.gov/tcgafiles/ftp_auth/distro_ftpusers/anonymous/tumor/",
#  "dataSubType":"copy number (gene-level)",
#  "security":"public",
#  "dataProducer":"UCSC Xena team",
#  "label":"gene-level copy number (gistic2_thresholded)",
#  "tags":
#  ["cancer", "non-small cell lung cancer", "gastric cancer",
#   "pancancer", "Adrenal gland", "Esophagus", "Rectum", "Endometrium",
#   "Brian", "Head and Neck region", "Thymus", "Lung", "Blood",
#   "Soft tissue", "Bone", "Stomach", "Colon and Rectum", "Prostate",
#   "Breast", "Ovary", "Kidney", "Lymphatic tissue", "Brain",
#   "Thyroid Gland", "Bladder", "Paraganglia", "Cervix", "Uterus",
#   "Liver", "Bile duct", "Eye", "Skin", "Colon", "Testis",
#   "Lining of body cavities", "White blood cell", "Pancreas"],
#  "path":
#  "/inside/home/jzhu/cgDataJing/TCGAscripts/preFreezeCAVM/TCGA/PANCAN/Gistic2_CopyNumber_Gistic2_all_thresholded.by_genes",
#  "anatomical_origin":"",
#  "name":"TCGA_PANCAN_gistic2_threshold",
#  "wrangling_procedure":
#  "Data is combined from all TCGA cohorts and deposited into UCSC Xena repository",
#  "sample_type":["tumor"],
#  "redistribution":true,
#  "type":"genomicMatrix",
#  "version":"2016-08-16",
#  "unit":
#  "-2,-1,0,1,2: 2 copy del,1 copy del,no change,amplification,high-amplification",
#  "primary_disease":"cancer",
#  "description":
#  "TCGA pan-cancer gene-level copy number variation (CNV) estimated using the GISTIC2 threshold method,compiled using data from all TCGA cohorts. Copy number was measured experimentally using whole genome microarray at a TCGA genome characterization center. Subsequently,GISTIC2 method was applied using the TCGA FIREHOSE pipeline to produce gene-level copy number estimates. GISTIC2 further thresholded the estimated values to -2,-1,0,1,2,representing homozygous deletion,single copy deletion,diploid normal copy,low-level copy number amplification,or high-level copy number amplification. Genes are mapped onto the human genome coordinates using UCSC cgData HUGO probeMap. Reference to GISTIC2 method PMID:21527027.<br><br>"}

```

--------------------------------------------------------------------------------


## LOAD TCGA PANCAN GENE FUSION DATA
```{r gene_fusions}
# Source: Driver Fusions and Their Implications in the Development 
#           and Treatment of Human Cancers
# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5916809/ (Table S1)

tcga_fusion <- readxl::read_xlsx("TCGA_DriverFusions_mmc2.xlsx", 
                                 sheet = 2, skip = 1) %>%
  rename(
    Study_Abbreviation = Cancer,
    Tumor_Sample_Barcode = Sample
  ) %>%
  mutate(
    participant_id = barcode_to_participant(Tumor_Sample_Barcode),
    sample_id = barcode_to_sample(Tumor_Sample_Barcode)
  ) %>%
  filter(sample_id %in% tcga_mut_samples) %>%
  filter(Spanning >= MIN_SPANNING_READ_COUNT) %>%
  filter(Junction >= MIN_JUNCTION_READ_COUNT) %>%
  select(Study_Abbreviation, participant_id, sample_id, Tumor_Sample_Barcode,
         everything()) %>%
  arrange(Study_Abbreviation, participant_id, sample_id, Tumor_Sample_Barcode)

```


--------------------------------------------------------------------------------

## CONSTRUCT SAMPLE GENOMIC VARIANT TABLE (SNV/INDEL, COPY ALT, GENE FUSION)
```{r sample_genomic_var_tbl}
genvar_mut <- tcga_maf %>%
  select(Study_Abbreviation, participant_id, sample_id, Tumor_Sample_Barcode,
         Hugo_Symbol, Variant_Classification, HGVSp_Short) %>%
  rename(
    TCGA_Cohort = Study_Abbreviation, 
    Participant_ID = participant_id, 
    Sample_ID = sample_id, 
    Gene = Hugo_Symbol,
    Variant_Type = Variant_Classification,
    Description = HGVSp_Short
  )

sample_to_cohort <- genvar_mut %>%
  select(Participant_ID, Sample_ID, TCGA_Cohort) %>%
  unique() %>%
  arrange(TCGA_Cohort, Participant_ID, Sample_ID)

copy_state_to_description <- c(
  "-2" = "homozygous_deletion",
  "-1" = "single_copy_loss",
  "1"  = "low_copy_gain",
  "2"  = "high_copy_gain"
)

genvar_cop <- tcga_cop %>%
  left_join(sample_to_cohort, by = c("sample_id" = "Sample_ID")) %>%
  mutate(
    Tumor_Sample_Barcode = NA,
    Variant_Type = "Copy_Number_Alteration",
    Description = copy_state_to_description[as.character(copy_state)]
  ) %>%
  rename(
    Gene = gene,
    Sample_ID = sample_id
  ) %>%
  select(colnames(genvar_mut))

genvar_fus <- tcga_fusion %>%
  rename(
    TCGA_Cohort = Study_Abbreviation,
    Participant_ID = participant_id,
    Sample_ID = sample_id,
    Gene = Fusion
  ) %>%
  mutate(
    Variant_Type = "Gene_Fusion",
    Description = Gene
  ) %>%
  select(colnames(genvar_mut))
  
  
genvar_tbl <- bind_rows(genvar_mut, genvar_cop, genvar_fus) %>%
  arrange(TCGA_Cohort, Participant_ID, Sample_ID, 
          Variant_Type, Gene, Description)

stopifnot(
  !any(is.na(genvar_tbl$TCGA_Cohort)),
  !any(is.na(genvar_tbl$Participant_ID)),
  !any(is.na(genvar_tbl$Sample_ID))
)

ptgenvar_tbl <- filter(genvar_tbl, Sample_ID == "TCGA-05-4244-01")

```

## SAMPLE VARIANT TYPE STATISTICS
```{r genvar_stats}
# Variant counts by type, per sample.
genvar_stats <- genvar_tbl %>%
  group_by(TCGA_Cohort, Sample_ID, Variant_Type) %>%
  summarize(Count = n()) %>%
  ungroup() %>%
  pivot_wider(names_from = "Variant_Type", values_from = "Count") %>%
  replace(is.na(.), 0)

genvar_median_counts <- genvar_stats %>%
  group_by(TCGA_Cohort) %>%
  summarize(
    Copy_Number_Alteration = median(Copy_Number_Alteration),
    Gene_Fusion = median(Gene_Fusion),
    Missense_Mutation = median(Missense_Mutation),
    Nonsense_Mutation = median(Nonsense_Mutation),
    Frame_Shift_Del = median(Frame_Shift_Del),
    Frame_Shift_Ins = median(Frame_Shift_Ins),
    Splice_Site = median(Splice_Site),
    In_Frame_Del = median(In_Frame_Del),
    In_Frame_Ins = median(In_Frame_Ins),
    Translation_Start_Site = median(Translation_Start_Site),
    Nonstop_Mutation = median(Nonstop_Mutation)
  ) %>%
  ungroup()

```


### OUTPUTS
```{r outputs}
readr::write_tsv(genvar_tbl, 
                 file = gzfile("TCGA_AllVarTypes_by_Sample.tsv.gz"))

writexl::write_xlsx(x = list(
    sample_variants_example = ptgenvar_tbl,
    sample_vartype_counts = genvar_stats,
    genvar_median_counts = genvar_median_counts
  ),
  path = "~/Downloads/TCGA_Sample_Variant_Table.xlsx"
)

```

